### DISCUSSION TITLE
How do you get it working with amd

### DISCUSSION THREAD

**Discussion by MaxOrtGit** [hlky/stable-diffusion#214] [Q&A]
When I run it I get the error:
RuntimeError: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx
Is there any way around it?

**Comment by LinuxForEveryone** [comment]
Try this guide? https://www.youtube.com/watch?v=d_CgaHyA_n4
My only question with this guide is how the heck do I use the WebUI, since this repo only has a script for Windows?

**Comment by cstueckrath** [comment]
As the scripts only work in Windows (and AMDs ROCm not) you have to do the heavy lifting yourself ;)

1. install conda
2. create and activate a venv (see below)
3. install pip
4. pip install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/rocm5.2/
5. run python scripts\webui.py or python scripts\relauncher.py
6. create a .sh with
```
conda env create -f environment.yaml
conda env update --file environment.yaml --prune
conda activate <environment name from environment.yaml>
python scripts/relauncher.py
```

Hope that works. I cannot test this myself at the moment, waiting for a new SSD.

**Comment by cstueckrath** [comment]
I tried to try this, but had some issues:

1. Don't try to run ROCm or anything AMD related on Ubuntu 22.04. The highest supported version is 20.04 at the moment.
2. make sure, that your GPU is supported. My 6700XT is not (you'll need a Navi 21 or better)
3. make sure that you actually installed all the rocm-stuff (sudo amdgpu-install --usecase=rocm) before
4. go here and use the script: https://github.com/hlky/stable-diffusion/wiki/Linux-Installation
5. activate lsd venv
6. pip uninstall torch related stuff
7. pip install torch from [nightly/rocm5.2/](https://download.pytorch.org/whl/nightly/rocm5.2/) (see above)
8. run "The Master Test" from https://rentry.org/sdamd (inside the activated venv) to check, if cuda is working

**Reply by phreeware** [reply]
> 2. make sure, that your GPU is supported. My 6700XT is not (you'll need a Navi 21 or better)

i know this is kinda off-topic but thanks on your inputs so far regarding windows and antares but this thread might be of interest to you: https://www.reddit.com/r/StableDiffusion/comments/ww436j/howto_stable_diffusion_on_an_amd_gpu/

**Reply by cstueckrath** [reply]
interesting, I'll check that out.
